creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -274,15 +291,112 @@ class SnapbackSM {
+   * @param {number} userId user id to issue a reconfiguration for
+   * @param {string} wallet wallet address of user id
+   * @param {string} primary endpoint of the current primary node on replica set
+   * @param {string} secondary1 endpoint of the current first secondary node on replica set
+   * @param {string} secondary2 endpoint of the current second secondary node on replica set
+   * @param {string[]} unhealthyReplicas array of endpoints of current replica set nodes that are unhealthy
+  */
+  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplicas) {
+    this.log(`[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} unhealthy replica set=[${unhealthyReplicas}]`)
+
+    const unhealthyReplicasSet = new Set(unhealthyReplicas)
+    const baseSyncRequestParams = { userWallet: wallet, syncType: SyncType.Manual }
+    let newReplicaSetSPIds = []
+
+    // If a primary is currently unhealthy or more than one node is unhealthy, skip for now
+    if (unhealthyReplicasSet.has(primary) || unhealthyReplicas.length > 1) {
+      this.log(`[issueUpdateReplicaSetOp] userId=${userId} skipping reconfig.`)
+      return
+    }
+
+    let phase = ''
+    try {
+      // Generate new replica set
+      phase = issueUpdateReplicaSetOpPhases.GET_HEALTHY_CONTENT_NODES
+      const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+      const newSecondary = await this.determineNewReplicaSet({ wallet, secondary1, secondary2, primary })
+
+      this.log(`[issueUpdateReplicaSetOp] Updating userId=${userId} wallet=${wallet} replica set=[${primary},${secondary1},${secondary2}] to new replica set=[${primary},${currentHealthySecondary},${newSecondary}]`)
+
+      if (!this.snapbackReconfigEnabled) return
+
+      // Create new array of replica set spIds and write to URSM
+      phase = issueUpdateReplicaSetOpPhases.UPDATE_URSM_REPLICA_SET
+      newReplicaSetSPIds = [
+        this.endpointToSPIdMap[primary],
+        this.endpointToSPIdMap[currentHealthySecondary],
+        this.endpointToSPIdMap[newSecondary]
+      ]
+
+      await this.audiusLibs.contracts.UserReplicaSetManagerClient.updateReplicaSet(
+        userId,
+        newReplicaSetSPIds[0], // primary
+        newReplicaSetSPIds.slice(1) // [secondary1, secondary2]
+      )
+
+      // Sync data from existing primary to new secondary
+      phase = issueUpdateReplicaSetOpPhases.ENQUEUE_SYNCS
+      await this.enqueueSync({
+        ...baseSyncRequestParams,
+        primaryEndpoint: primary,
+        secondaryEndpoint: newSecondary
+      })
+    } catch (e) {
+      const errorMsg = `[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} failed at phase=${phase} reconfiguring to spIds=[${newReplicaSetSPIds}]: ${e.toString()}\n${e.stack}`
+      this.logError(errorMsg)
+      throw new Error(errorMsg)
```

Comment:
by re-throwing this error, the logic on lines 693 would short circuit all reconfig work on a single failed reconfig. i think its fine to throw error, but we should handle that error externally and adjust the `numUpdateReplicaOpsIssued` tracking

Reply:
handling this flow such that one issue reconfig op does not stop the entire flow